src/Microsoft.Azure.WebJobs.Host.Storage/Singleton/BlobLeaseDistributedLockManager.cs

```diff
@@ -167,10 +173,15 @@ protected virtual BlobContainerClient GetParentBlobContainerClient(BlobClient bl
             bool blobDoesNotExist = false;
             try
             {
-                // Optimistically try to acquire the lease. The blob may not yet
-                // exist. If it doesn't we handle the 404, create it, and retry below
-                var leaseResponse = await GetBlobLeaseClient(blobClient, proposedLeaseId).AcquireAsync(leasePeriod, cancellationToken: cancellationToken);
-                return leaseResponse.Value.LeaseId;
+                // Check if a lease is available before trying to acquire.
+                // If it doesn't we handle the 404, create it, and retry below
+                var blobProperties = GetBlobProperties(blobClient);
+
+                if (blobProperties?.LeaseState != LeaseState.Leased)
```

Comment:
Should probably consider other states and if they're safe to attempt to acquire on

https://docs.microsoft.com/en-us/rest/api/storageservices/lease-blob#remarks